### PR TITLE
Fixing repo_id regex validation error on windows platforms

### PR DESCRIPTION
**Pull request by Mystfit**
# What does this PR do?

Fixes bug introduced in [#3746](https://github.com/huggingface/diffusers/pull/3746).

On windows platforms, the use of os.path.join introduces backslashes into `repo_id` and `file_path`. This causes the `REPO_ID_REGEX` regex in `_validators.py` to not match the namespace and model page from the URL correctly. Here's a [regex validator example](https://regex101.com/r/4Rx1fl/2) which demonstrates how backslashes and forward slashes get matched.

This commit checks if we are using a local file path or a URL by checking if we removed the huggingface website prefix earlier and if true, we use the old `"/".join()` style of creating repo_id and file_path with forward slashes. If we're dealing with a file path then os.path.join will use backslashes on windows platforms instead.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/diffusers/blob/main/CONTRIBUTING.md)?
- [x] Did you read our [philosophy doc](https://github.com/huggingface/diffusers/blob/main/PHILOSOPHY.md) (important for complex PRs)?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/diffusers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

@patrickvonplaten

### CODE REVIEW

File: src/diffusers/loaders.py
Comment (patrickvonplaten) [inline src/diffusers/loaders.py:1899]:
```suggestion
            file_path = "/".join(ckpt_path.parts[2:]) if is_URL else os.path.join(*ckpt_path.parts[2:])
```

Comment (patrickvonplaten) [reply src/diffusers/loaders.py:1899]: @Mystfit this should be corrected no? Apart from this this PR looks good to be merged to me :-)

**Comment by patrickvonplaten** [comment]
Happy to merge after!

File: src/diffusers/loaders.py
Comment (iamwavecut) [reply src/diffusers/loaders.py:1899]: @Mystfit up

Comment (iamwavecut) [reply src/diffusers/loaders.py:1899]: @patrickvonplaten as you're the code owner, you should merge that suggestion because, obviously, it's a typo.

**Comment by sayakpaul** [comment]
@Mystfit let us know when you would like another review. 

Cc: @yiyixuxu too.

**Comment by Mystfit** [comment]
@patrickvonplaten I think this is ready to commit. I just did one last merge from main but code quality checks are failing from an unrelated commit.

File: src/diffusers/loaders.py
Comment (iamwavecut) [reply src/diffusers/loaders.py:1908]: No, u didn't, look carefully at else part on 1908: it should start from 2 and to the end. 
```suggestion
            file_path = "/".join(ckpt_path.parts[2:]) if is_URL else os.path.join(*ckpt_path.parts[2:])
```

Comment (Mystfit) [reply src/diffusers/loaders.py:1908]: > No, u didn't, look carefully at else part on 1908: it should start from 2 and to the end.

I've just realised something with the latter part of that line. If we're not dealing with a URL and are using a file, then why are we slicing the path from the second position onwards? Wouldn't this make no sense for a file path?

**Review by iamwavecut — APPROVED** [review]

**Comment by iamwavecut** [comment]
@sayakpaul @patrickvonplaten @yiyixuxu, Please take a look, Windows link loading is broken in `main` and it's kinda blocker 😢

**Review by sayakpaul — APPROVED** [review]
LGTM

@yiyixuxu could you also give this a look? 

@Mystfit we need to run `make style && make quality` to make sure the CI is green.

**Review by yiyixuxu — APPROVED** [review]
great! thank you!

**Pull request merged**
